XiEditor/XiCore.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Protocol describing the frontend interface with core.
+/// Documentation for the protocol can be found here:
+/// https://xi-editor.github.io/xi-editor/docs/frontend-protocol.html
+protocol XiCore {
```

Comment:
This protocol should probably be `: class`, since we assign references to it in multiple places.